### PR TITLE
Stake unstake tests

### PR DESCRIPTION
**Pull request by Kishan-Dhakan**
- Lock tokens in stake pool
- Check wallet balance decreases
- Check stake pool information to find the stake pool, confirm client ID and balance
- Unlock tokens
- Check wallet balance increase
- Check stake pool info is not retaining deleted pool

### CODE REVIEW

File: tests/cli_tests/zboxcli_stake_unstake_token_test.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+		require.Nil(t, err, "Error unmarshalling stake pool info", strings.Join(output, "\n"))
+
+		delegates := stakePool.Delegate
+
+		// Pool Id returned by sp-lock should be found in sp-info with correct balance
+		found := false
+		for _, delegate := range delegates {
+			if delegate.ID == stakePoolID {
+				found = true
+				require.Equal(t, int64(5000000000), delegate.Balance)
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_stake_unstake_token_test.go:76]: add a message to explain what this assertion means if it fails

File: tests/cli_tests/zboxcli_stake_unstake_token_test.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+				require.Equal(t, wallet.ClientID, delegate.DelegateID)
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_stake_unstake_token_test.go:77]: add a message to explain what this assertion means if it fails

File: tests/cli_tests/zboxcli_stake_unstake_token_test.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+		err = json.Unmarshal([]byte(output[0]), &stakePool)
+		require.Nil(t, err, "Error unmarshalling stake pool info", strings.Join(output, "\n"))
+
+		delegates = stakePool.Delegate
+
+		// Pool Id returned by sp-lock should be found in sp-info with correct balance
+		found = false
+		for _, delegate := range delegates {
+			if delegate.ID == stakePoolID {
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_stake_unstake_token_test.go:111]: add logging

**Review by avanaur — COMMENTED** [review]
Is the following to be addressed here?
1. Various param conditions for `./zbox sp-lock` and `./zbox sp-unlock` (eg. tokens > balance, negative tokens, etc.)
2. The earnings from staking

**Pull request merged**